components/nimbus/src/enrollment.rs

```diff
@@ -373,40 +375,75 @@ impl ExperimentEnrollment {
                 enrollment_id,
                 branch,
                 ..
-            } => EnrollmentChangeEvent::new(
-                &self.slug,
-                enrollment_id,
-                branch,
-                None,
-                EnrollmentChangeEventType::Enrollment,
-            ),
+            } => {
+                nimbus_events::enrollment.record(nimbus_events::EnrollmentExtra {
+                    branch: Some(branch.clone()),
+                    enrollment_id: Some(enrollment_id.clone().to_string()),
```

Comment:
nit: do we need the `clone` here? I would of thought `to_string()` is enough as that probably allocates a new string anyways

(and ditto for below)

Reply:
As a tiny follow-up we file, I notice we have to clone a bunch of times, when the `branch` and `enrollment_id` values don't really change

but that makes sense since the generated glean metrics code expects `String`s... can we file a ticket/issue against glean/glean-metrics to allow for strings to be reference, or just use a [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html)

Reply:
I am probably a little over-zealous with `clone()` as a way to get along with the borrow checker, I'll try your suggestions to see if I can make this better, but at least these aren't large objects so I wasn't too worried about cloning them.